proposed/2022/vulnerabilities-in-restore.md

```diff
@@ -80,6 +80,14 @@ This feature will be opt-in to start and gather feedback from developers.
 
 To enable the feature, a developer can add `<NuGetAudit>enable</NuGetAudit>` to their project file as a MSBuild property. To disable the feature, a developer can add `<NuGetAudit>disable</NuGetAudit>` or remove the property from the project file.
 
+#### Setting Vulnerability Auditing Modes
+
+There will be different modes to audit vulnerabilities based on the developer's or developer's team preference. To do this, a developer will opt-in to a feature called `<NuGetAuditMode>` which will have different modes such as `direct`, and `all`.
+
+These modes should be pretty straight-forward. `direct` will scan for any top-level vulnerabilities, and `all` will scan for both top-level and transitive-level vulnerabilities. The default will be `direct` until the experience is ready to be `all` given that transitive vulnerabilities are the majority of vulnerability notices (90%+).

```

Comment:
What are your thoughts on adding a message to the output along the lines of: "Only direct dependencies were scanned for vulnerabilities. To enable scanning transitive dependencies as well, turn on <NuGetAudit>all</NuGetAudit>."

Reply:
Some customers are actively hostile against any non-actionable message that can't also be suprressed. If we has Roslyn analyzer style `.editorconfig` seveity levels on codes, then we could allow customers to customize their builds as they like. But currently we only have NU codes for warnings and errors, informational messages don't have codes and therefore can't be suppressed.

So, it's a tradeoff, especially in large solutions where the message could be shown hundreds of times, once for each solution.

One problem with moving fast, is that we don't have time to carefully plan, or test, or validate, before we have to ship. So, my recommendation for now is no, don't have a message, and we can wait for customer feedback while working on other improvements, and if we don't get feedback, we can do a bit of our own testing to estimate how important it is.